### PR TITLE
Create workflow component

### PR DESCRIPTION
**Pull request by balzdur**
Implement core components + local state for workflow edition

### CODE REVIEW

File: packages/app-builder/src/components/ReactFlow.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    <ControlButton
+      title="Automatic layout"
+      onClick={() => {
+        layoutElem({ fitView: false });
+      }}
+    >
+      <Icon icon="tree-schema" />
+    </ControlButton>
+  );
+}
```
Comment (balzdur) [inline packages/app-builder/src/components/ReactFlow.tsx:81]: I started to factorize some global utility concerning `react-flow`

File: packages/app-builder/src/components/Scenario/Workflow/DetailPannel/AddToCaseIfPossibleNode.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+              >
+                <Highlight text={scenario.name} query={searchValue} />
+              </SelectWithCombobox.ComboboxItem>
+            );
+          })}
+        </SelectWithCombobox.ComboboxList>
+      </SelectWithCombobox.Popover>
+    </SelectWithCombobox.Root>
+  );
+}
```
Comment (balzdur) [inline packages/app-builder/src/components/Scenario/Workflow/DetailPannel/AddToCaseIfPossibleNode.tsx:115]: This is the specific implementation for a detail pannel corresponding to a specific node. You can read only one of those to understand how it works (others are kindly the same)

File: packages/app-builder/src/components/Scenario/Workflow/DetailPannel/DetailPannel.tsx [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+function ActionNode({ id, data }: { id: string; data: ActionData }) {
+  switch (data.type) {
+    case 'create-case':
+      return <CreateCaseNode id={id} data={data} />;
+    case 'add-to-case-if-possible':
+      return <AddToCaseIfPossibleNode id={id} data={data} />;
+    default:
+      assertNever('Unknown action data type', data);
+  }
+}
```
Comment (balzdur) [inline packages/app-builder/src/components/Scenario/Workflow/DetailPannel/DetailPannel.tsx:236]: This file is the "global" implentation of the `DetailPannel` : it display the corresping node depending on the sate and business rules.

File: packages/app-builder/src/components/Scenario/Workflow/Nodes/AddToCaseIfPossibleActionContent.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          i18nKey="workflows:action_node.add_to_case_if_possible.content"
+          components={{
+            Inbox: <span className="font-bold" />,
+          }}
+          values={{ inbox: selectedInbox.name }}
+        />
+      </p>
+    </div>
+  );
+}
```
Comment (balzdur) [inline packages/app-builder/src/components/Scenario/Workflow/Nodes/AddToCaseIfPossibleActionContent.tsx:44]: This is a specific node content implementation. 

Nodes are all the same :
- a shared "header" (used to bring rapid information corresponding to the node)
- a specific content sumarry (like this file)

File: packages/app-builder/src/components/Scenario/Workflow/WorkflowFlow.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+      }
+
+      return {
+        ...nd,
+        position,
+      };
+    }),
+    edges: edges,
+  };
+}
```
Comment (balzdur) [inline packages/app-builder/src/components/Scenario/Workflow/WorkflowFlow.tsx:139]: The basic shell for `react-flow` really similar to what we already have for data model (-> this is were we inject specific settings like node types, auto-layout function...)

File: packages/app-builder/src/components/Scenario/Workflow/WorkflowProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+  }
+  if (nodes.some((node) => isTriggerData(node.data))) {
+    return 'action';
+  }
+  return 'trigger';
+}
+
+export function useWorkflowActions() {
+  return useWorkflowStore((state) => state.actions);
+}
```
Comment (balzdur) [inline packages/app-builder/src/components/Scenario/Workflow/WorkflowProvider.tsx:251]: This file is were the internal state logic is implemented :
- internal state + actions (= zustand)
- external data required to display contextual information like scenario, inboxes names (= Classic React Context)

File: packages/app-builder/public/locales/en/workflows.json [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  "action_node.create_case.empty_content": "Select missing information to complete the configuration.",
+  "action_node.create_case.content": "Create a new case in the inbox <Inbox>{{inbox}}</Inbox>",
+  "action_node.add_to_case_if_possible.title": "Add to an existing case",
+  "action_node.add_to_case_if_possible.entity": "case",
+  "action_node.add_to_case_if_possible.empty_content": "Select missing information to complete the configuration.",
+  "action_node.add_to_case_if_possible.content": "Add to an open matching case in the inbox <Inbox>{{inbox}}</Inbox>. If no matching case is found, a new case will be created.",
+  "detail_pannel.no_selected_nodes.title": "Workflow detail",
+  "detail_pannel.no_selected_nodes.description": "Select a node to see its details.",
+  "detail_pannel.multiple_selected_nodes.title": "Multiple selected nodes",
+  "detail_pannel.multiple_selected_nodes.description": "You have selected multiple nodes. This panel don't show details of multiple nodes. Please select only one node to see details.",
```
Comment (Pascal-Delange) [inline packages/app-builder/public/locales/en/workflows.json:18]:
```suggestion
  "detail_pannel.multiple_selected_nodes.description": "You have selected multiple nodes. This panel doesn't show details of multiple nodes. Please select only one node to see details.",
```

File: packages/app-builder/public/locales/en/workflows.json [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  "detail_pannel.multiple_selected_nodes.clear_selection": "Clear selection",
+  "detail_pannel.create_trigger_node.title": "Choose a trigger",
+  "detail_pannel.create_trigger_node.description": "A trigger is an event that starts the workflow. Choose a trigger to start the workflow.",
+  "detail_pannel.create_action_node.title": "Choose an action",
+  "detail_pannel.create_action_node.description": "An action is a task that the workflow performs. Choose an action to add to the workflow.",
+  "detail_pannel.decision_created.description": "Trigered when a decision is created which matches the following criteria.",
```
Comment (Pascal-Delange) [inline packages/app-builder/public/locales/en/workflows.json:24]:
```suggestion
  "detail_pannel.decision_created.description": "Triggered when a decision is created, that matches the following criteria.",
```

**Pull request merged**